aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java

```diff
@@ -34,40 +35,42 @@
 
 public class AssumeRoleAwsClientFactory implements AwsClientFactory {
   private AwsProperties awsProperties;
+  private HttpClientProperties httpClientProperties;
+  private S3FileIOProperties s3FileIOProperties;
```

Comment:
it seems those are never assigned, which needs to happen in `initialize(..)`:
```
--- a/aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java
+++ b/aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java
@@ -78,6 +78,8 @@ public class AssumeRoleAwsClientFactory implements AwsClientFactory {
   @Override
   public void initialize(Map<String, String> properties) {
     this.awsProperties = new AwsProperties(properties);
+    this.s3FileIOProperties = new S3FileIOProperties(properties);
+    this.httpClientProperties = new HttpClientProperties(properties);

```

Reply:
Good catch, I initialized in other classes but forgot this one. I'm fixing.